### PR TITLE
Make all Databases use the ReportFilepath class 

### PR DESCRIPTION
**Pull request by Sinerum**
Fix databases by fully incorparating the new ReportFilepath interface.

### CODE REVIEW

**Review by boehmseb — CHANGES_REQUESTED** [review]
There are still a lot of `ReportFilepath` related type errors remaining.
Here an excerpt of the mypy output (mypy 0.991):

```
varats/varats/plots/bug_fixing_relations.py:501: error: Argument 1 to "SZZUnleashedReport" has incompatible type "ReportFilepath"; expected "Path"  [arg-type]
varats/varats/paper_mgmt/case_study.py:687: error: Argument 1 has incompatible type "ReportFilepath"; expected "Path"  [arg-type]
varats/varats/tables/time_stats_table.py:36: error: Argument 1 to "TimeReportAggregate" has incompatible type "ReportFilepath"; expected "Path"  [arg-type]
varats/varats/tables/global_eval.py:143: error: Argument 1 to "filter_report_paths_binary" has incompatible type "List[ReportFilepath]"; expected "List[Path]"  [arg-type]
varats/varats/tables/global_eval.py:158: error: Argument 1 to "filter_report_paths_binary" has incompatible type "List[ReportFilepath]"; expected "List[Path]"  [arg-type]
varats/varats/tables/feature_analysis_eval.py:119: error: Argument 1 to "filter_report_paths_binary" has incompatible type "List[ReportFilepath]"; expected "List[Path]"  [arg-type]
varats/varats/tables/feature_analysis_eval.py:278: error: Argument 1 to "filter_report_paths_binary" has incompatible type "List[ReportFilepath]"; expected "List[Path]"  [arg-type]
varats/varats/paper_mgmt/paper_config_manager.py:137: error: Incompatible return value type (got "List[ReportFilepath]", expected "List[Path]")  [return-value]
varats/varats/data/databases/commit_interaction_database.py:87: error: Argument 1 to "ReportFilename" has incompatible type "ReportFilepath"; expected "Union[str, Path]"  [arg-type]
varats/varats/data/databases/blame_verifier_report_database.py:165: error: Argument 1 to "ReportFilename" has incompatible type "ReportFilepath"; expected "Union[str, Path]"  [arg-type]
varats/varats/data/databases/blame_verifier_report_database.py:165: error: "ReportFilepath" has no attribute "name"  [attr-defined]
varats/varats/data/databases/blame_library_interactions_database.py:53: error: Argument 1 to "load_blame_report" has incompatible type "ReportFilepath"; expected "Path"  [arg-type]
varats/varats/data/databases/blame_library_interactions_database.py:112: error: Argument 1 to "ReportFilename" has incompatible type "ReportFilepath"; expected "Union[str, Path]"  [arg-type]
varats/varats/data/databases/blame_interaction_degree_database.py:250: error: Argument 1 to "ReportFilename" has incompatible type "ReportFilepath"; expected "Union[str, Path]"  [arg-type]
varats/varats/data/databases/blame_interaction_database.py:51: error: Argument 1 to "load_blame_report" has incompatible type "ReportFilepath"; expected "Path"  [arg-type]
varats/varats/data/databases/blame_interaction_database.py:88: error: Argument 1 to "ReportFilename" has incompatible type "ReportFilepath"; expected "Union[str, Path]"  [arg-type]
varats/varats/data/databases/blame_diff_metrics_database.py:113: error: Argument 1 to "ReportFilename" has incompatible type "ReportFilepath"; expected "Union[str, Path]"  [arg-type]
varats/varats/data/databases/blame_diff_metrics_database.py:113: error: Value expression in dictionary comprehension has incompatible type "ReportFilepath"; expected type "Path"  [misc]
varats/varats/data/databases/blame_diff_metrics_database.py:123: error: Argument 1 to "ReportFilename" has incompatible type "ReportFilepath"; expected "Union[str, Path]"  [arg-type]
varats/varats/data/databases/blame_diff_metrics_database.py:123: error: Value expression in dictionary comprehension has incompatible type "ReportFilepath"; expected type "Path"  [misc]
varats/varats/data/databases/blame_diff_metrics_database.py:394: error: Argument 1 to "id_from_paths" has incompatible type "Tuple[Path, Path]"; expected "Tuple[ReportFilepath, ReportFilepath]"  [arg-type]
varats/varats/data/databases/blame_diff_metrics_database.py:395: error: Argument 1 to "timestamp_from_paths" has incompatible type "Tuple[Path, Path]"; expected "Tuple[ReportFilepath, ReportFilepath]"  [arg-type]
varats/varats/data/databases/blame_diff_metrics_database.py:407: error: Argument 7 to "build_cached_report_table" has incompatible type "Callable[[Tuple[ReportFilepath, ReportFilepath]], str]"; expected "Callable[[Tuple[Path, Path]], str]"  [arg-type]
varats/varats/data/databases/blame_diff_metrics_database.py:407: error: Argument 8 to "build_cached_report_table" has incompatible type "Callable[[Tuple[ReportFilepath, ReportFilepath]], str]"; expected "Callable[[Tuple[Path, Path]], str]"  [arg-type]
varats/varats/data/databases/blame_diff_library_interaction_database.py:116: error: Argument 6 to "build_cached_report_table" has incompatible type "Callable[[Tuple[ReportFilepath, ReportFilepath]], Tuple[Any, str, str]]"; expected "Callable[[Tuple[Path, Path]], Tuple[Any, str, str]]"  [arg-type]
varats/varats/data/databases/blame_diff_library_interaction_database.py:117: error: Argument 7 to "build_cached_report_table" has incompatible type "Callable[[Tuple[ReportFilepath, ReportFilepath]], str]"; expected "Callable[[Tuple[Path, Path]], str]"  [arg-type]
varats/varats/data/databases/blame_diff_library_interaction_database.py:117: error: Argument 8 to "build_cached_report_table" has incompatible type "Callable[[Tuple[ReportFilepath, ReportFilepath]], str]"; expected "Callable[[Tuple[Path, Path]], str]"  [arg-type]
```

File: varats/varats/data/databases/blame_interaction_degree_database.py [hunk 1]
```diff
@@ -97,9 +96,9 @@ def create_dataframe_layout() -> pd.DataFrame:
             return df_layout
 
         def create_data_frame_for_report(
-            report_path: Path
+            report_path: ReportFilepath
         ) -> tp.Tuple[pd.DataFrame, str, str]:
-            report = load_blame_report(report_path)
+            report = load_blame_report(report_path.full_path())
```
Comment (vulder) [inline varats/varats/data/databases/blame_interaction_degree_database.py]: Would it then not also make sense to port all `load_*_report` functions to `ReportFilepath`?

Comment (Sinerum) [reply varats/varats/data/databases/blame_interaction_degree_database.py]: Yes, that has now been changed.

File: varats/varats/jupyterhelper/file.py [hunk 2]
```diff
@@ -21,9 +21,12 @@
     PyDrillerSZZReport,
 )
 from varats.mapping.commit_map import CommitMap
+from varats.report.report import ReportFilepath
+
+PathLikeTy = tp.TypeVar('PathLikeTy', Path, ReportFilepath)
```
Comment (vulder) [inline varats/varats/jupyterhelper/file.py]:
```suggestion
```
You can directly import the type

File: varats/varats/jupyterhelper/file.py [hunk 3]
```diff
@@ -1,6 +1,6 @@
 """This module provides different jupyther helpers to allow easier interaction
 with varas file handling APIs."""
-
+import typing as tp
 from pathlib import Path
 
 from varats.data.data_manager import VDM
```
Comment (vulder) [inline varats/varats/jupyterhelper/file.py]:
```suggestion
from varats.data.data_manager import VDM, PathLikeTy
```

**Review by vulder — APPROVED** [review]

**Review by boehmseb — APPROVED** [review]
Once that last pylint issue is fixed, we can merge this.

**Pull request merged**